src/java/org/apache/cassandra/service/accord/FetchTopology.java

```diff
@@ -122,19 +114,26 @@ public Response(long epoch, Topology topology)
     public static final IVerbHandler<FetchTopology> handler = message -> {
         long epoch = message.payload.epoch;
         Topology topology = AccordService.instance().topology().maybeGlobalForEpoch(epoch);
-        if (topology == null)
-            MessagingService.instance().respond(Response.UNKNOWN, message);
-        else
+        if (topology != null)
             MessagingService.instance().respond(new Response(epoch, topology), message);
+        else
+            throw new IllegalStateException("Unknown topology: " + epoch);
```

Comment:
said this in another thread, but given the refactor its not showing up in the file, so reposing.

I do not think this is a good idea.  `UNKNOWN` failure is truly unknown... it could be a NPE, it could be we ran *before* we were ready... it could be anything... its unknown... I do not think its a good idea to conflate that with "I don't know this epoch", because the handling is different...


if `UNKNOWN` exception is returned, retrying on the same node makes sense, it could be an ephemeral issue.
If the node doesn't know about the epoch, calling them again isn't the best idea... we should try another node, and if no node knows the epoch we are in a bad state...

Reply:
Switched to specialized UNKNOWN_TOPLOGY failure response.